### PR TITLE
Add runtime data_queue

### PR DESCRIPTION
**Pull request by junjiejiangjjj**
col-based storage with row-based interface

Signed-off-by: junjie.jiang <junjie.jiang@zilliz.com>

### CODE REVIEW

File: towhee/runtime/data_queue.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            self._not_empty.notify_all()
+
+    @property
+    def sealed(self) -> bool:
+        return self._sealed
+
+
+class ColumnType(Enum):
+    QUEUE = auto()
+    NOQUEUE = auto()
```
Comment (GuoRentong) [inline towhee/runtime/data_queue.py:104]: ARRAY?

File: towhee/runtime/data_queue.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+# limitations under the License.
+
+from typing import List, Tuple
+import threading
+from enum import Enum, auto
+
+from collections import deque, namedtuple
+
+
+class DataQueue:
```
Comment (GuoRentong) [inline towhee/runtime/data_queue.py:22]: why can't we directly wrap one of the Queue types in python queue module?

**Review by GuoRentong — CHANGES_REQUESTED** [review]

File: towhee/runtime/data_queue.py (hunk 2 above)
Comment (junjiejiangjjj) [reply towhee/runtime/data_queue.py:22]: Queue is a thread-safe wrapper of deque

File: towhee/runtime/data_queue.py (hunk 1 above)
Comment (junjiejiangjjj) [reply towhee/runtime/data_queue.py:104]: NOQUEUE   means only having one cell,  a list, an int, or others.

File: towhee/runtime/data_queue.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        for col_type in self._schema.col_types():
+            if col_type == ColumnType.QUEUE:
+                self._data.append(_QueueColumn())
+            else:
+                self._data.append(_NoQueueColumn())
+            
+        self._sealed = False
+        self._size = 0
+        self._lock = threading.Lock()
+        self._not_full = threading.Condition(self._lock)
```
Comment (filip-halt) [inline towhee/runtime/data_queue.py:37]: by using these conditions wont we require a seperate thread for every queue?

Comment (junjiejiangjjj) [reply towhee/runtime/data_queue.py:37]: yeah, each node of the graph will have a thread

**Pull request merged**